### ISSUE TITLE
PyLint false positive PLW0602 when `del` a global

### ISSUE THREAD

**Issue by ilius** [astral-sh/ruff#10397]
Sample code:
```
test_global_var = None


def delete_test_global_var():
	global test_global_var
	if "test_global_var" in globals():
		del test_global_var

```

Command: `ruff check --preview --isolated --select PL del-global.py`
Output:
```
del-global.py:5:9: PLW0603 Using the global statement to update `test_global_var` is discouraged
  |
4 | def delete_test_global_var():
5 |     global test_global_var
  |            ^^^^^^^^^^^^^^^ PLW0603
6 |     if "test_global_var" in globals():
7 |         del test_global_var
  |

del-global.py:5:9: PLW0602 Using global for `test_global_var` but no assignment is done
  |
4 | def delete_test_global_var():
5 |     global test_global_var
  |            ^^^^^^^^^^^^^^^ PLW0602
6 |     if "test_global_var" in globals():
7 |         del test_global_var
  |

Found 2 errors.

```

Version:
```
$ ruff version
ruff 0.3.2
```

**Comment by charliermarsh** [comment]
Thanks!

**Comment by charliermarsh** [comment]
This only occurs when the `del` is in a conditional.
